CONTRIBUTING.md

```diff
@@ -0,0 +1,95 @@
+
+1. Point out decisions you made and what alternatives you considered. If you're unsure about a particular approach, ask a question to make your own thinking clear and help the reviewer identify controversial parts of the proposed solution. For example: "here I returned a result object to represent the status after the transaction. I also considered throwing an exception in case of error but I didn't like that it made it unclear where an error happened. Which do you prefer and why?" This is particularly important for [work in progress pull requests](#work-in-progress-pull-requests).
+
+1. Follow the guidelines below.
+
+## The review process
+Bug fixes, pull requests corresponding to issues with a clearly stated goal and pull requests with clear tests and/or process for manual verification are given priority. Pull requests that are unclear or controversial may be tagged as `needs discussion` and/or may take longer to review.
+
+We try to have at least two people review every pull request and we encourage everyone to participate in the review process to get familiar with the code base and help ensure higher quality. Reviewers should ask themselves some or all of the following questions:
+- Was this change adequately discussed prior to implementation?
+- Is the intended behavior clear under all conditions?
+- What interesting cases should be verified?
+- Is the behavior as intended in all cases?
+- What other functionality could this PR affect? Does that functionality still work as intended?
+- Was the change verified with several different devices and Android versions?
+- Is the code easy to understand and to maintain?
+- Is there sufficient detail to inform any changes to documentation?
+
+When a pull request is first created, @opendatakit-bot tags it as `needs review` to indicate that code review is needed. Community members review the code and leave their comments, verifying that the changes included are relevant and properly address the issue. A maintainer does a thorough code review and when satisfied with the code, tags the pull request as `needs testing` to indicate the need for a manual [black-box testing](https://en.wikipedia.org/wiki/Black-box_testing) pass. A pull request may go back and forth between `needs testing` and `needs review` until the behavior is thoroughly verified. Once the behavior has been thoroughly verified, the pull request is tagged as `behavior verified`. A maintainer then merges the changes. Pull requests that need more complete reviews including review of approach and/or appropriateness are tagged with `reviews wanted`. Any community member is encouraged to participate in the review process!
+
+Small fixes that target very particular bugs may occasionally be merged without a second review.
+
+## Work in progress pull requests
+
+Work in progress (WIP) pull requests are useful to illustrate a proposed direction and get early feedback before committing to a solution direction.
```

Comment:
Now that GitHub itself has a feature to make a pull request a work in progress, I think we should add that bit as well